xmtp_db/src/encrypted_store/identity.rs

```diff
@@ -80,7 +86,10 @@ impl<C: ConnectionExt> DbConnection<C> {
                 .first::<Option<i64>>(conn)
         })?;
 
-        Ok(matches!(next_rotation_opt, Some(rotate_at) if now_ns() >= rotate_at))
+        Ok(match next_rotation_opt {
+            Some(rotate_at) => now_ns() >= rotate_at,
+            None => true,
```

Comment:
this will cover the cases where we didn't have the lifetime as rotation time. cover old users.